uswds/uswds-site
# USWDS-Site - Snyk: Add postcss override to package.json

# Summary

Added an override for postcss in the `package.json` file to prevent a snyk error.

> **Note**
> This override should be removed once the vulnerability can be resolved in @uswds/compile v1.1.0

## Problem statement
Running `npx snyk test` results in the following errror: 
```
Issues with no direct upgrade or patch:
  ✗ Improper Input Validation [Medium Severity][https://security.snyk.io/vuln/SNYK-JS-POSTCSS-5926692] in postcss@8.4.19
    introduced by @uswds/compile@1.0.0 > gulp-sourcemaps@3.0.0 > @gulp-sourcemaps/identity-map@2.0.1 > postcss@7.0.39 and 1 other path(s)
  This issue was fixed in versions: 8.4.31
```

This is a vulnerability related to a subdependency of [gulp-sourcemaps](https://www.npmjs.com/package/gulp-sourcemaps), which has has not been updated for 3 years. Because of the  dependency's slow update cycle, it is unlikely that this vulnerability will be resolved by `gulp-sourcemaps`.

## Solution
Creating a temporary override for this sub-dependency will resolve the snyk error while we wait for an update to uswds/compile. 

Alternatively, we could update the snyk ignore, but that would require periodic updates every 30 days and would not resolve the vulnerability. 

## Testing and review
1. Confirm that there are no snyk errors related to postcss
2. Confirm that creating an override is the best method for resolving this error
